src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
 }
 
 
+/*
+ * keeper_ensure_pg_configuration_files_in_pgdata checks if
+ * postgresql.conf, pg_hba.conf, pg_ident.conf files exist
+ * in $PGDATA, if not it tries to get them from default
+ * location and modifies paths inside copied postgresql.conf.
+ */
+static bool
+keeper_ensure_pg_configuration_files_in_pgdata(KeeperConfig *config)
+{
+	char pgdata[MAXPGPATH];
+	char hbaConfPath[MAXPGPATH];
+	char posgresqlConfPath[MAXPGPATH];
+	char *pgdataRelativePath = NULL;
+	char defaultConfigurationPath[MAXPGPATH];
+	bool configurationPathPresent = false;
+	bool postgresqlConfExists = false;
+	char defaultPosgresqlConfPath[MAXPGPATH];
+	char defaultHbaConfPath[MAXPGPATH];
+	char defaultIdentConfPath[MAXPGPATH];
+	char identConfPath[MAXPGPATH];
+
+	get_absolute_path(config->pgSetup.pgdata, pgdata, MAXPGPATH);
```

Comment:
PGDATA is supposed to exist on-disk already. When it does, we've been using `realpath` to get the absolute path to PGDATA in other places. See `pg_setup_set_absolute_pgdata` in `pgsetup.c` for instance. Why not using the same approach here, or maybe even checking that `pg_setup_set_absolute_pgdata` has been called already (and adding a comment about it), or calling it from here?

Reply:
used `realpath`